paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/FlinkTableSource.java

```diff
@@ -149,9 +149,18 @@ protected Integer inferSourceParallelism(StreamExecutionEnvironment env) {
                     Boolean.parseBoolean(envConfig.toMap().get(FLINK_INFER_SCAN_PARALLELISM)));
         }
         Integer parallelism = options.get(FlinkConnectorOptions.SCAN_PARALLELISM);
-        if (parallelism == null && options.get(FlinkConnectorOptions.INFER_SCAN_PARALLELISM)) {
+        if (parallelism == null
```

Comment:
I'm trying to understand the parallelism inferring logic this PR wants to achieve, please see if this is correct.

1. If Paimon option SCAN_PARALLELISM is configured, then use this value.
2. Else if Flink configuration `parallelism.default` is configured, then use this value.
3. Else if Paimon option INFER_SCAN_PARALLELISM is set to true, and the parallelism of source can be inferred by Paimon (like when fixed bucket + unbounded stream), then Paimon would provide an inference result.
4. Else Paimon would not set the parallelism of the source operator, instead it would be Flink infra that decide the parallelism of this operator